conformance/lang/expressions/type_cast_expression.balt

```diff
@@ -0,0 +1,2012 @@
+    io:println(typeof d); // @output typedesc record {| readonly int a; "Car" b; |}
+
+    int[] e = <[int, byte]>[444, 21];
+    io:println(typeof e); // @output typedesc [int,byte]
+
+    int[]|byte[] f = <[int, byte]>[444, 21];
+    io:println(typeof f); // @output typedesc [int,byte]
+
+    any g = <float>1;
+    io:println(g); // @output 1.0
+
+    any h = <decimal>1.4;
+    io:println(h.toBalString()); // @output 1.4d
+}
+
+Test-Case: output
+Description: Test cast to readonly can be used to cause constructors withing the expression to construct values
+             with the read-only bit on.
+Labels: any, array-type, list-constructor-expr, map-type, mapping-constructor-expr, readonly-type, string, 
+        table-constructor-expr, table-type, type-cast-expr, typeof-expr
+
+function init() {
+    int[] a = <readonly>[1, 2];
+    io:println(typeof a); // @output typedesc [1,2]
+
+    any[] b = <readonly>[1, "Ballerina"];
+    io:println(typeof b); // @output typedesc [1,"Ballerina"]
+
+    map<string> c = <readonly>{a: "4", b: "car"};
+    io:println(typeof c); // @output typedesc {"a":"4","b":"car"}
+
+    map<any> d = <readonly>{a: 3, b: "car"};
+    io:println(typeof d); // @output typedesc {"a":3,"b":"car"}
+
+    table<record {|int a;|}> e = <readonly>table [
+            {a: 54}
+    ];
+    io:println(typeof e); // @output typedesc [{"a":54}]
+}
+
+Test-Case: output
+Description: Test cast to readonly can be used to cause constructors withing the expression to construct values
+             with the read-only bit on.
+Labels: array-type, byte, byte-array-literal, readonly-type, type-cast-expr, typeof-expr, var
+        
+function init() {
+    byte[1] f = <readonly>base16 `aa`;
+    io:println(typeof f); // @output typedesc [170]
+
+    var g = <byte[] & readonly>base16 `aa 4b`;
+    io:println(typeof g); // @output typedesc [170,75]
+
+    byte[2] h = <byte[2] & readonly>base16 `ba16`;
+    io:println(typeof h); // @output typedesc [186,22]
+
+    var i = <int[1] & readonly>base16 `ff`;
+    io:println(typeof i); // @output typedesc [255]
+
+    byte[3] j = <readonly>base64 `AAAB`;
+    io:println(typeof j); // @output typedesc [0,0,1]
+
+    var k = <byte[] & readonly>base64 `AAAB ABB=`;
+    io:println(typeof k); // @output typedesc [0,0,1,0,16]
+
+    byte[] m = <byte[3] & readonly>base64 `pqrs`;
+    io:println(typeof m); // @output typedesc [166,170,236]
+
+    var n = <int[3] & readonly>base64 `+/12`;
+    io:println(typeof n); // @output typedesc [251,253,118]
+}
+
+Test-Case: output
+Description: Test cast to readonly can be used to verify that the value resulting from the evaluation of expression 
+             has its read-only bit on. 
+Labels: any, array-type, list-constructor-expr, map-type, mapping-constructor-expr, readonly-type, string, 
+        table-constructor-expr, table-type, type-cast-expr, typeof-expr
+
+function init() {
+    int[] a1 = <readonly>[1, 2];
+    int[] a = <readonly>a1;
```

Comment:
This tests two things: 1. constructing a value with a readonly cast, 2. verifying that the value is immutable at runtime using the cast. Only the second one matches the description. I would do

```ballerina
int[] & readonly a = [1, 2];
int[] b = a;
int[] c = <readonly> b;
```